### PR TITLE
Put mod_jingle_sip options in a map with defaults

### PR DESCRIPTION
**Pull request by gustawlippa**
Put options in a map with defaults.
Added a possibility to configure options that were used in the code, and documented these new options. However, these are not tested yet, and from what I understand, without host types the username-phone mapping won't work great.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/115836) / small_tests / fa30cf846efc7a3d2b044409c153efc2e69411aa
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3600/115836/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3600/115836/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-21_14.49.50/lib.mongooseim.logs/run.2022-03-21_14.49.51/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/115911) / small_tests / 4837817acf180febfa9ec4180e22dc3afa657547
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3600/115911/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3600/115911/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-21_16.39.42/lib.mongooseim.logs/run.2022-03-21_16.39.42/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/116038) / small_tests / bb5a3d5b29b7e8f7a091daec79aae0ad4af8cf33
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3600/116038/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3600/116038/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-22_09.33.23/lib.mongooseim.logs/run.2022-03-22_09.33.24/suite.log.html)

File: doc/modules/mod_jingle_sip.md [hunk 1]
```diff
@@ -137,6 +137,20 @@ The value used to create SIP URIs (including VIA headers).
...
+* **Syntax:** string
+* **Default:** `"udp"`
+* **Example:** `transport = "tcp"`
+
+The SIP transport parameter used when calling the proxy.
+
+### `modules.mod_jingle_sip.username_to_phone`
+* **Syntax:** Array of TOML tables with the following keys: `username` and `phone`, and string values
+* **Default:** `[]`
+* **Example:** `username_to_phone = [{username = "2000006168", phone = +919177074440}]`
```
Comment (chrzaszcz) [inline doc/modules/mod_jingle_sip.md:150]:
```suggestion
* **Example:** `username_to_phone = [{username = "2000006168", phone = "+919177074440"}]`
```

File: src/jingle_sip/mod_jingle_sip.erl [hunk 2]
```diff
@@ -360,9 +380,9 @@ make_user_header({User, _} = US) ->
 
 
 get_proxy_uri(Server) ->
-    ProxyHost = gen_mod:get_module_opt(Server, ?MODULE, proxy_host, "localhost"),
-    ProxyPort = gen_mod:get_module_opt(Server, ?MODULE, proxy_port, 5060),
-    Transport = gen_mod:get_module_opt(Server, ?MODULE, transport, "udp"),
+    ProxyHost = gen_mod:get_module_opt(Server, ?MODULE, proxy_host),
+    ProxyPort = gen_mod:get_module_opt(Server, ?MODULE, proxy_port),
+    Transport = gen_mod:get_module_opt(Server, ?MODULE, transport),
```
Comment (chrzaszcz) [inline src/jingle_sip/mod_jingle_sip.erl:385]: Just a thought: how about getting the opts with one lookup: `gen_mod:get_loaded_module_opts(Server, ?MODULE)` and then pattern-matching the map?

**Review by chrzaszcz — COMMENTED** [review]
Looks good, I only added minor comments.

File: src/jingle_sip/mod_jingle_sip.erl (hunk 2 above)
Comment (gustawlippa) [reply src/jingle_sip/mod_jingle_sip.erl:385]: Yeah, seems like a good idea, I'll see how it looks

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/116421) / small_tests / c0f8f9261a22e7dd27ed14d5e2f17524f915b3c7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3600/116421/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3600/116421/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-23_15.47.58/lib.mongooseim.logs/run.2022-03-23_15.47.59/suite.log.html)

File: src/jingle_sip/jingle_sip_helper.erl [hunk 3]
```diff
@@ -45,9 +45,9 @@ jingle_iq(ToBinary, FromBinary, JingleEl) ->
 
 -spec maybe_rewrite_to_phone(mongoose_acc:t()) -> jid:jid().
 maybe_rewrite_to_phone(Acc) ->
-    Server = mongoose_acc:lserver(Acc),
+    HT = mongoose_acc:host_type(Acc),
```
Comment (arcusfelis) [inline src/jingle_sip/jingle_sip_helper.erl:48]: Can we use HostType variable name instead?

**Review by arcusfelis — APPROVED** [review]

**Review by chrzaszcz — APPROVED** [review]

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/117101) / small_tests / e17c1b601c0db69bc11022ee2969ce7a35aa10f0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3600/117101/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3600/117101/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-28_08.51.41/lib.mongooseim.logs/run.2022-03-28_08.51.41/suite.log.html)

**Pull request merged**